### ISSUE TITLE
grid search result API

### ISSUE THREAD

**Issue by adrinjalali** [probabl-ai/skore#715]
In the intro example we have:

```py
# %%
import pandas as pd

df = pd.DataFrame(gs_cv.cv_results_)
df.insert(len(df.columns), "rmse", -df["mean_test_score"].values)
df[["param_alpha", "rmse"]].head()

# %%
import matplotlib.pyplot as plt

fig = plt.figure(layout="constrained", dpi=200)
plt.plot(df["param_alpha"], df["rmse"])
plt.xscale("log")
plt.xlabel("Alpha hyperparameter")
plt.ylabel("RMSE")
plt.title("Ridge regression")
plt.show()

# %%
# |
# Finally, we store some relevant information to our skore project, so that we
# can visualize them later in the skore UI for example:

# %%
my_project.put("my_gs_cv", gs_cv)
my_project.put("my_df", df)
my_project.put("my_fig", fig)
```

I would have expected this to be more like:

```py
my_project.pu("GS Results", gs_cv.cv_results_, type="cv_results")
```

kinda thing which then does all the magic for me. Also, we don't really need to translate the `"neg_root_mean_squared_error"`.

**Comment by adrinjalali** [comment]
you can always do

```py
def put(_type, / , **kwargs):
```

**Comment by rouk1** [comment]
Related to #731

**Comment by adrinjalali** [comment]
yes, plot and processing of the data

**Comment by MarieS-WiMLDS** [comment]
That's interesting. It's also a feedback from the user we saw on Friday: it's a pain having to rewrite all the time the same code to display similar plots.  
There is library called [dabl](https://github.com/dabl/dabl), which tries to do that, but it's not in a very lively phase. It's a idea we have in mind.

**Comment by adrinjalali** [comment]
Yes, I've told @tuscland  about `dabl` a while ago. It was Andy (Andreas Müller)'s idea, and it went pretty nicely till he moved to Microsoft. dabl has a lot of good ideas.

**Comment by tuscland** [comment]
> Side-note: the hypothetical type argument [...]

Like in W&B, we use objects to define a type, with automatic coercion for basic data structure we recognize. If we are able to get the cross validation result to look like something we know how to handle, then there is no need to explicitly specify a type.

I'd like to explore how we can extend the automatic wrapping functionality to `*SearchCV` classes (like GridSearchCV, RandomizedSearchCV). Similar to `cross_validate`, this would allow us to compute additional metrics that weren't initially specified, even after the search/training is complete.

**Comment by tuscland** [comment]
Can you elaborate a bit why this is a bad idea? Not trying to convince you, but I need to understand.

**Comment by adrinjalali** [comment]
Sorry I should have explained myself better.

I like explicit API, but I also understand that we don't want to make users to write too much boilerplate code. That's why I don't like a method that changes the default scorers calculated and stored, but I like the idea of making it very easy for people to have them calculated, post facto to the grid search itself. Since those metrics are not relevant to the choice of what hyperparameters to choose, we don't need them during the search anyway. However, having a nice tool which makes it easy to have them calculated after the fact, makes total sense to me.

This also avoids computation which is not necessary during the search itself.

And generally, it gives the power back to the user, while they can still do the easy stuff really easily, and we let them do more complicated stuff if they want to.

Hope this explains my position a bit.

**Comment by GaelVaroquaux** [comment]
> FWIW, I think automatically calculating extra metrics (and having wrappers in the first place) is a bad idea

As a manager, a recurrent and major problem that I face is that my data scientists run very long computation not storing the right information. I would like skore to store as much information as possible, even when misused. Let us not assume that our users do things right, they often don't
